crates/sui-core/src/node_sync/node_state.rs

```diff
@@ -371,69 +343,113 @@ where
         }
     }
 
-    async fn process_exec_driver_digest(
+    async fn get_true_effects(
         &self,
-        permit: OwnedSemaphorePermit,
-        digest: &TransactionDigest,
-    ) -> SuiResult {
-        trace!(?digest, "validator pending execution requested");
+        cert: &CertifiedTransaction,
+    ) -> SuiResult<SignedTransactionEffects> {
+        let digest = cert.digest();
+        match self.node_sync_store.get_effects(digest)? {
+            Some(effects) => Ok(effects),
+            None => {
+                let effects = self.aggregator.execute_cert_to_true_effects(cert).await?;
```

Comment:
The `execute_cert_to_true_effects` API in AuthorityAggregator is a bit strange to me. It returns the last signed effect when validity is reached. The signature is probably useless. Raw effects should be sufficient.
Similarly we should just store raw effects in the effects table.

Reply:
Agreed, will fix. I just did it this way because it was the most convenient thing.